### PR TITLE
Consolidate Submit and Toxic Exposure feature flags

### PR DESCRIPTION
**Pull request by sethdarragile6**
This PR consolidates the `disability_compensation_lighthouse_submit_migration` flipper flag into the `disability_526_toxic_exposure` flipper flag. This simplifies deployment of Toxic Exposure, which can only be supported by Lighthouse (not EVSS). And since the `disability_compensation_lighthouse_submit_migration` is NOT used by the front end, while `disability_526_toxic_exposure` IS... hello the former, bye bye the latter 👋

## Summary

- *This work is behind a feature toggle (flipper): YES _and_ NO*
This change IS feature toggle. It IS flipper 🐬
- *(Summarize the changes that have been made to the platform)*
Swapped out all instances of `disability_compensation_lighthouse_submit_migration` with `disability_526_toxic_exposure`, and adjusted tests accordingly
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?)*
Disability Benefits Experience Team 1 (DbEx-tReX 🦖)
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
Consolidate /submit and Toxic Exposure feature flag#[83405](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/83405)
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by tblackwe — APPROVED** [review]
Reminder: the flipper will need to removed from the database once this is merged into prod

File: app/sidekiq/evss/disability_compensation_form/submit_form526.rb
Comment (freeheeling) [inline app/sidekiq/evss/disability_compensation_form/submit_form526.rb:110]: I think this could be further consolidated by also removing the Flipper evaluation. If the corresponding feature was ever enabled for a user, their associated in progress form data would be appended with the `includeToxicExposure` indicator, which signifies the setting of the `submit_endpoint` to 'claims_api' when the submission is created.

**Review by freeheeling — APPROVED** [review]
Approved with a suggestion for further consolidation.

**Review by aurora-a-k-a-lightning — DISMISSED** [review]

File: app/sidekiq/evss/disability_compensation_form/submit_form526.rb
Comment (tblackwe) [reply app/sidekiq/evss/disability_compensation_form/submit_form526.rb:110]: agreed. When I proposed the `includeToxicExposure` element,  the expectation was that the flipper would be used in the IPF creation, and then the `includeToxicExposure` indicator would be used going forward.

Comment (sethdarragile6) [reply app/sidekiq/evss/disability_compensation_form/submit_form526.rb]: Your collective confidence is inspiring- I have removed it! Please take a quick look (seems like I need re-approval anyway. Thanks! @freeheeling @tblackwe @aurora-a-k-a-lightning

File: spec/sidekiq/evss/disability_compensation_form/submit_form526_all_claim_spec.rb
Comment (aurora-a-k-a-lightning) [inline spec/sidekiq/evss/disability_compensation_form/submit_form526_all_claim_spec.rb:496]: whoa, hold on. we still need this test.
having the `submit_endpoint: 'claims_api'` will make this pass, right? 🤔 
we just don't need the flipper enabling/disabling parts

**Review by aurora-a-k-a-lightning — CHANGES_REQUESTED** [review]
waaaait!

**Review by tblackwe — APPROVED** [review]

**Review by freeheeling — COMMENTED** [review]
I'd suggest renaming the context description on [L470](https://github.com/department-of-veterans-affairs/vets-api/pull/17078/files#diff-5637d19f3d91d39ba3c0b35006091a43aac2adb5ba3313ece91a326f3b830e65R470:~:text=for%20user%27%20do-,context%20%27with%20toxic%20exposure%20Flipper%20enabled%20for%20user%27%20do,let(%3Asubmission)%20do,-let(%3A))

**Review by freeheeling — APPROVED** [review]
lgtm

**Review by aurora-a-k-a-lightning — APPROVED** [review]
lgtm!

**Review by ryan-mcneil — APPROVED** [review]
:shipit:

**Pull request merged**
